### PR TITLE
fix: solve publicly shared document with hide download issue

### PR DESCRIPTION
**Pull request by luka-nextcloud**
Signed-off-by: Luka Trovic <luka@nextcloud.com>

### 📝 Summary

* Resolves: #3704

#### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
B | A

### 🚧 TODO

- [ ] ...

### 🏁 Checklist

- [ ] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [ ] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [ ] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

**Comment by juliushaertl** [comment]
Tests need adjustment

File: cypress/e2e/share.spec.js [hunk 1]
```diff
@@ -122,17 +122,4 @@ describe('Open test.md in viewer', function() {
 				cy.get('.modal-mask').should('not.exist')
 			})
 	})
-
-	it('Share a file with download disabled shows an error', function() {
```
[the remark this answers is not in the document]
Comment (luka-nextcloud) [reply cypress/e2e/share.spec.js:126]: I have a concern. I set the download attribute to false for a share but the download button still appears. Is there any additional thing I need to do?

[the remark this answers is not in the document]
Comment (juliushaertl) [reply cypress/e2e/share.spec.js:126]: Let's not remove this test.

For clarity. There are two separate features:
- Disable download (on internal shares only) which will block the download of the file -> For this we need to show the error message that this cypress test checks for
- Hide download (only on share links) which does not block and just hides any visible download options -> Which the change in DocumentService of this PR brings back to work.

[the remark this answers is not in the document]
Comment (juliushaertl) [reply cypress/e2e/share.spec.js:126]: Cypress still pass fine with the change locally, but are a bit flaky due to other reasons.

**Review by juliushaertl — CHANGES_REQUESTED** [review]

File: cypress/e2e/share.spec.js [hunk 2]
```diff
@@ -122,17 +122,4 @@ describe('Open test.md in viewer', function() {
...
-		cy.shareFileToUser('test.md', recipient, {
```
[the remark this answers is not in the document]
Comment (max-nextcloud) [reply cypress/e2e/share.spec.js:127]: @luka-nextcloud this is still missing. Just a matter of reverting the file to the state in main.

**Comment by luka-nextcloud** [comment]
@max-nextcloud @juliushaertl It's all done now. :)

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**